src/sage/combinat/k_regular_sequence.py

```diff
@@ -1204,14 +1223,37 @@ def guess(self, f, n_max=100, max_dimension=10, sequence=None):
         zero = domain(0)
         one = domain(1)
 
+        # A `line` will be a triple `(t, r, s)` corresponding to an entry
+        # `k**t * m + r` belonging to `M_s`
+        # TODO: what is `M_s`?
+
+        # The elements of `lines` will be correspond to the current components
+        # of the left vector valued sequence described in the algorithm section
+        # of the docstring.
+
         def values(m, lines):
+            """
+                Return current (as defined by ``lines``) left vector valued
+                sequence for argument ``m``.
+            """
             return tuple(seq(m)) + tuple(f(k**t_R * m + r_R) for t_R, r_R, s_R in lines)
 
         @cached_function(key=lambda lines: len(lines))  # we assume that existing lines are not changed (we allow appending of new lines)
         def some_inverse_U_matrix(lines):
+            r"""
+                Find an invertible `d \times d` times submatrix of the matrix
```

Comment:
```suggestion
                Find an invertible `d \times d` submatrix of the matrix
```

Reply:
Copied to https://github.com/sagemath/sage/pull/35682/files#r1208506621 .